install.sh

```diff
@@ -1,22 +1,82 @@
 #!/bin/sh
 set -e
-# Docker CE for Linux installation script
+# Docker Engine for Linux installation script.
 #
-# See https://docs.docker.com/engine/install/ for the installation steps.
+# This script is intended as a convenient way to configure docker's package
+# repositories and to install Docker Engine, This script is not recommended
+# for production environments. Before running this script, make yourself familiar
+# with potential risks and limitations, and refer to the installation manual
+# at https://docs.docker.com/engine/install/ for alternative installation methods.
 #
-# This script is meant for quick & easy install via:
-#   $ curl -fsSL https://get.docker.com -o get-docker.sh
-#   $ sh get-docker.sh
+# - The script requires `root` or `sudo` privileges to run.
+# - The script attempts to detect your Linux distribution and version and
+#   configure your package management system for you.
+# - The script doesn't allow you to customize most installation parameters.
+# - The script installs dependencies and recommendations without asking for
+#   confirmation.
+# - By default, the script installs the latest stable release of Docker CLI,
+#   Docker Engine, Docker Buildx, Docker Compose, containerd, and runc. When
+#   using this script to provision a machine, this may result in unexpected
+#   major version upgrades of these packages. Always test upgrades in
+#   a test environment before deploying to your production systems.
+# - The script isn't designed to upgrade an existing Docker installation. When
+#   using the script to update an existing installation, dependencies may not be
+#   updated to the expected version, resulting in outdated versions.
```

Comment:
nit: "The script" is repeated in every bullet sentence. It could be extracted into a list header:

```suggestion
# The script:
# - Requires `root` or `sudo` privileges to run.
# - Attempts to detect your Linux distribution and version and
#   configure your package management system for you.
# - Doesn't allow you to customize most installation parameters.
# - Installs dependencies and recommendations without asking for
#   confirmation.
# - Installs the latest stable release (by default) of Docker CLI,
#   Docker Engine, Docker Buildx, Docker Compose, containerd, and runc. When
#   using this script to provision a machine, this may result in unexpected
#   major version upgrades of these packages. Always test upgrades in
#   a test environment before deploying to your production systems.
# - Isn't designed to upgrade an existing Docker installation. When
#   using the script to update an existing installation, dependencies may not be
#   updated to the expected version, resulting in outdated versions.
```